modules/utxo-coredao/src/descriptor.ts

```diff
@@ -0,0 +1,47 @@
+import { BIP32Interface, bitgo } from '@bitgo/utxo-lib';
+
+/**
+ * Script type for a descriptor. This is either a p2sh (sh) or a p2sh-p2wsh (sh-wsh) script.
+ */
+export type ScriptType = 'sh' | 'sh-wsh';
+
+/**
+ * Create a multi-sig descriptor to produce a coredao staking address
+ * @param scriptType segwit or legacy
+ * @param locktime locktime for CLTV
+ * @param m Total number of keys required to unlock
+ * @param orderedKeys
+ */
+export function createMultiSigDescriptor(
+  scriptType: ScriptType,
+  locktime: number,
+  m: number,
+  orderedKeys: BIP32Interface[]
+): string {
+  if (m > orderedKeys.length || m < 1) {
+    throw new Error(
+      `m (${m}) must be less than or equal to the number of keys (${orderedKeys.length}) and greater than 0`
+    );
+  }
+  if (locktime <= 0) {
+    throw new Error(`locktime (${locktime}) must be greater than 0`);
+  }
+
+  const xpubs = orderedKeys.map((key) => key.toBase58() + '/*');
```

Comment:
should we default to neutering these keys unless explicitly said? Would protect

Reply:
sounds good